cx/spark/server.js

```diff
@@ -0,0 +1,208 @@
+});
+
+process.on('SIGTERM', () => {
+  listener.close(async ()=>{
+    console.log('Closing http server.');
+    await deleteWebhooks();
+    process.exit(0);
+  });
+});
+
+async function init(){
+  await deleteWebhooks();
+  registerWebhook();
+}
+
+// Converts Spark message to a detectIntent request. 
+function sparkToDetectIntent(message, sessionPath){
+     const request = {
+     session: sessionPath,
+     queryInput: {
+         text: {
+             text: message.text,
+            },
+            languageCode,
+        },
+    };
+
+  return request;
+}
+
+// Converts detectIntent response to a Spark text message. 
+function detectIntentToSparkMessage(response){
+    agentResponse = '';
+    
+    for (const message of response.queryResult.responseMessages) {
+        if (message.text) {
+            agentResponse += `${message.text.text}\n`;
+        };
+    };
+  
+  if(agentResponse.length != ''){
+    const request = {
+      text: agentResponse
+     };
+    return request;
+    };
+};
+
+/**
+ * This function calls Dialogflow CX API to retrieve the response
+ * https://cloud.google.com/dialogflow/cx/docs/quick/api
+ */
+async function detectIntentText(message, personId) {
+  const sessionId = personId;
+  const sessionPath = client.projectLocationAgentSessionPath(
+      projectId,
+      locationId,
+      agentId,
+      sessionId
+  );
+  console.info(sessionPath);
+
+  request = sparkToDetectIntent(message, sessionPath);
+  const response = await client.detectIntent(request);
+
+  return response;
+}
+
+function sendMessage(text, personEmail) {
+  request.post('https://api.ciscospark.com/v1/messages', {
+    auth: {
+      bearer: sparkAccessToken
+    },
+    json: {
+      "toPersonEmail": personEmail,
+      "text": text
+    }
+  }, (err, resp, body) => {
+    if (err) {
+      console.error('Failed to send message :' + err);
+    }
+  });
+}
+
+function registerWebhook() {
+  request.post('https://api.ciscospark.com/v1/webhooks', {
+    auth: {
+      bearer: sparkAccessToken
+    },
+    json: {
+      "name": "test",
+      "targetUrl": targetUrl,
+      "resource": "messages",
+      "event": "created"
+    }
+  }, (err, resp, body) => {
+    if (err) {
+      console.error('Failed to create Webhook :' + err);
+    }
+  });
+}
+
+function deleteWebhooks() {
+  return new Promise((resolve, reject) =>{
+    request.get('https://api.ciscospark.com/v1/webhooks?max=100', {
+      auth: {
+        bearer: sparkAccessToken
+      }
+    }, (err, resp, body) => {
+      if (err) {
+        console.error('Failed to check webhooks :' + err);
+        reject();
+      }
+      var webhooks = JSON.parse(resp.body).items;
+      if (Array.isArray(webhooks)) {
+        webhooks = webhooks.filter((value, index, arr)=> {
+          return value.targetUrl===targetUrl;
+        });
+        webhooks.forEach((webhook) => {
+          request.delete(
+              'https://api.ciscospark.com/v1/webhooks/' +
```

Comment:
@tyhu-google   From my understanding it uses the DELETE method here on the list of messages

Reply:
kk, so actually we want to delete all webhooks. That makes sense. Thanks for your clarification..